homeassistant/components/upb/scene.py

```diff
@@ -42,6 +50,31 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class UpbLink(UpbEntity, Scene):
     """Representation of an UPB Link."""
 
+    @property
+    def device_info(self):
+        """Device info for the entity."""
+        return {
+            "name": self._element.name,
+            "identifiers": {(DOMAIN, self._element.index)},
+            "manufacturer": "Generic",
+            "model": "UPB PIM",
+        }
+
+    def _element_changed(self, element, changeset):
+        change = changeset.get("last_change")
+        if change is None:
+            return
+        if change.get("command") is None:
+            return
+        if change["command"] not in TRIGGER_TYPES:
+            return
+
+        event = f"{DOMAIN}.scene_{change['command']}"
+        data = {ATTR_ENTITY_ID: self.entity_id}
+        data[ATTR_BRIGHTNESS_PCT] = change.get("level", -1)
+        data[ATTR_RATE] = change.get("rate", -1)
+        self.hass.bus.fire(event, data)
```

Comment:
@MartinHjelmare Before I run though this review, is firing the event in this context still OK?